src/RegularizedMoments.cc

```diff
@@ -0,0 +1,689 @@
+    firstTruth << 0,
+                  0,
+                  0,
+                  0.310466905407061,
+                  -0.00373831312952513,
+                  0.0000112532002694914;
+
+    if (abs(zeroTruth - zeroRes) > tol) {
+        return false;
+    }
+    return approxEqual(firstRes, firstTruth, tol);
+}
+
+bool testBetaXY(double tol) {
+    Moments Q, W;
+    std::tie(Q, W) = buildTestMoments();
+    double zeroRes = BetaXY::computeValue(Q, W);
+    double bottom = makeBetaGradDivisor(Q, W);
+    Moments firstRes = BetaXY::computeGradient(Q, W, bottom);
+    double zeroTruth = 0.543777255954378;
+    Moments firstTruth;
+    firstTruth << 0,
+                  0,
+                  0,
+                  0.0205607222123882,
+                  0.261745049520270,
+                  -0.00157657335775577;
+
+    if (abs(zeroTruth - zeroRes) > tol) {
+        return false;
+    }
+    return approxEqual(firstRes, firstTruth, tol);
+}
+
+bool testBetaY(double tol) {
+    Moments Q, W;
+    std::tie(Q, W) = buildTestMoments();
+    double zeroRes = BetaY::computeValue(Q, W);
+    double bottom = makeBetaGradDivisor(Q, W);
+    Moments firstRes = BetaY::computeGradient(Q, W, bottom);
+    double zeroTruth = 1.91680644079168;
+    Moments firstTruth;
+    firstTruth << 0,
+                  0,
+                  0,
+                  0.00136163723260849,
+                  0.0346846138706271,
+                  0.220877927421585;
+
+    if (abs(zeroTruth - zeroRes) > tol) {
+        return false;
+    }
+    return approxEqual(firstRes, firstTruth, tol);
+}
+
+
+bool testConstant(double tol) {
+    Moments Q, W;
+    std::tie(Q, W) = buildTestMoments();
+    double zeroRes = Constant::computeValue(Q, W);
+    Moments firstRes = Constant::computeGradient(Q, W);
+    double zeroTruth = 0.0915084542604366/afw::geom::PI;
+    Moments firstTruth;
+    firstTruth << 0,
+                  -0.000675339145833491/afw::geom::PI,
+                  0.00138137552556848/afw::geom::PI,
+                  -0.0118159430257175/afw::geom::PI,
+                  0.00674319680500958/afw::geom::PI,
+                  -0.00689645127785071/afw::geom::PI;
+
+    if (abs(zeroTruth - zeroRes) > tol) {
+        return false;
+    }
+    return approxEqual(firstRes, firstTruth, tol);
+}
+
+// Sets the location in moment space to evaluate
+void MomentsModel::at(Moments const & inputQ) {
+   Q = Moments(inputQ); 
+   makeValue();
+}
+
+// Calculates the value of the weighted biased moments
+void MomentsModel::makeValue() {
+    alpha = makeAlpha(Q, W);
+    beta = makeBeta(Q, W);
+    constant = Constant::computeValue(Q, W);
+
+    double zero = Q[0]*constant;
+    FirstMoment one = zero*alpha;
+    SecondMoment two = zero*(beta + alpha*alpha.transpose());
+
+    value << zero, one[0], one[1], two(0, 0), two(0, 1), two(1, 1);
+}
+
+// Returns the value of the weighted biased moments calculation
+Moments MomentsModel::computeValues() {
+    return value;
+}
```

Comment:
This doesn't compute anything. Also why is `value` singular and `computeValues` plural? Perhaps `getValue` would be better?

Reply:
change value to plural. Also this signature was chosen so in the future someone my substitute a different objective function where the work may be done in the computeValues function, and the api will stay the same